scripts/simplify_network.py

```diff
@@ -354,37 +375,80 @@ def split_links(nodes):
                 n, busmap, costs, config, connection_costs_per_link, buses
             )
 
-            all_links = [i for _, i in sum(links, [])]
+            # TODO revise a variable name for `links` variable
+            # `links` is a list which contains dc-relevant graph elements
+            all_dc = [i for _, i in sum(links, [])]
+            all_links = list(set(n.links.index).intersection(all_dc))
+            all_dc_lines = list(set(n.lines.index).intersection(all_dc))
+
+            # p_max_pu = config["links"].get("p_max_pu", 1.0)
+            all_dc_lengths = pd.concat(
+                [n.links.loc[all_links, "length"], n.lines.loc[all_dc_lines, "length"]]
+            )
+            name = all_dc_lengths.idxmax() + "+{}".format(len(all_dc) - 1)
+
+            # HVDC part is represented as "Link" component
+            if dc_as_links:
+                p_max_pu = config["links"].get("p_max_pu", 1.0)
+                lengths = n.links.loc[all_links, "length"]
+                length = sum(
+                    n.links.loc[[i for _, i in l if _ == "Link"], "length"].mean()
+                    for l in links
+                )
+                p_nom = min(
+                    n.links.loc[[i for _, i in l if _ == "Link"], "p_nom"].sum()
```

Comment:
The list `[i for _, i in l if _ == "Link"]` is duplicated, it may be worth to add a variable and use it here, especially if it reduces the number of lines with the pre-commit

Reply:
Done :)